typed_python/_runtime.cpp

```diff
@@ -303,20 +340,139 @@ extern "C" {
         return floorresult;
     }
 
-    PyObject* nativepython_runtime_int_to_pyobj(int64_t i) {
+    // returns layout*
+    uint8_t *np_runtime_pyobj_to_typed(PyObject *obj, uint8_t* tgt, Type* tp) {
+        // tgt is a layout pointer
+        PyEnsureGilAcquired acquireTheGil;
+        PyInstance::copyConstructFromPythonInstance((Type *)tp, (instance_ptr)&tgt, obj, true);
```

Comment:
you need to be passing tgt, not `&tgt`.  Imagine Type is NamedTuple(x=int,y=int). This takes up 16 bytes. As it currently stands, you will take the NamedTuple out of the python object and jam it into the 16 bytes of stackspace that have been allotted for `tgt`, which is 8 bytes. 

This function ought to read

```
void np_runtime_pyobj_to_typed(PyObject* ob, instance_ptr tgt, Type* tp) {
    PyEnsureGilAcquired acquireTheGil;
    PyInstance::copyConstructFromPythonInstance(tp, tgt, obj, true);
}
```

Then, when you call it, just drop the 'store' operation.  Take the NamedTuple case. You're already passing it 'targetInstance.expr' , which is in fact a NamedTuple*. You cast it to void* to pass it to this function, and it comes in as 'tgt'.